python/cloudtik/providers/_private/aliyun/config.py

```diff
@@ -380,6 +380,9 @@ def _configure_ram_role_for_head(config):
     # Add IAM role to "head_node" field so that it is applied only to
     # the head node -- not to workers with the same node type as the head.
     config["head_node"]["RamRoleName"] = head_instance_ram_role_name
+    # Add IAM role to "provider" filed so that node_provider for head
+    # can construct client by ecs_ram_role credential
+    config["provider"]["ram_role_name"] = head_instance_ram_role_name
```

Comment:
There is a risk of potential issue here. "ram_role_name" in provider will be saved in the cluster config both local on working machine and head config. But "ram_role_name" is invalid for local machine and only valid for head. This may make your get_credential function mistakenly use ram_role_name on working node if there is no aliyun_credential (but we want to use something in the environment)

I would suggest to move this code to prepare_config_for_head in node provider just like we do for Azure to set the managed client id.

prepare_for_head_node:
ram_role_name = config.get("head_node", {}).get("RamRoleName")
if ram_role_name :
    remote_config["provider"]["ram_role_name "] = ram_role_name 

so that ram_role_name attribute will only appear at head node config.

Reply:
@ Thanks for your suggestion! I've refined this code.